File: src/main/java/Extensions.java
import java.util.Optional;

public class Extensions {

    public static void main(String[] args) {
        Optional<String> optional = Optional.of("Some value");
//        optional.is
        ExtensionsKt.ifPresentOrElse(optional, System.out::println, () -> System.out.println("absent"));
        //TODO - Invoke extension function in Java
    }
}

File: src/main/kotlin/extensions.kt
import java.util.*
import java.util.function.Consumer

fun <T> Optional<T>.ifPresentOrElse(consumer: Consumer<T>, runnable: Runnable) {
    //TODO Implement method
    if (this.isPresent ) {
        consumer.accept(this.get())
    } else {
        runnable.run()
    }
}

fun main() {
//    val optional: Optional<String> = Optional.of("value")
    val optionalEmpty: Optional<String> = Optional.empty()
    optionalEmpty.ifPresentOrElse(Consumer { println(it) }, Runnable{ println("Absent") })
}